tests/pytests/test_cursors.py

```diff
@@ -365,35 +365,113 @@ def next_cursor_command():
             for i in range(n_docs):
                 env.assertContains(i, result_set)
 
+# MOD-8483
+# Upon timeout, the sorter switches to yield mode until its heap is depleted.
+# Before the fix, the timeout flag was not reset after depleting the heap, causing subsequent FT.CURSOR READ
+# commands to always return empty results without depleting the cursor.
+# After the fix, the accumulated results until the timeout are returned, and the cursor is properly depleted.
+def testCursorDepletionNonStrictTimeoutPolicySortby():
+    env = Env(protocol=3, moduleArgs='ON_TIMEOUT RETURN')
+    conn = getConnectionByEnv(env)
+
+    # Create the index
+    env.expect('FT.CREATE idx SCHEMA n numeric').ok()
+
+    # Populate the index
+    num_docs = 150 * env.shardsCount
+    for i in range(num_docs):
+        conn.execute_command('HSET', f'doc{i}' ,'n', i)
+
+    starting_cursor_count = getCursorStats(env, 'idx')['index_total']
+
+    # Create a cursor that will timeout during accumulation of results
+    timeout_res_count = 3
+    cursor_count = 5
+    res, cursor = runDebugQueryCommandTimeoutAfterN(env, ['FT.AGGREGATE', 'idx', '*', 'sortby', '1', '@n', 'WITHCURSOR', 'count',
+                          cursor_count], timeout_res_count)
+    VerifyTimeoutWarningResp3(env, res)
+
+    # Verify that the accumulated results (up to timeout_res_count) are returned after timeout
+    env.assertEqual(len(res['results']), timeout_res_count)
+    n_received = len(res['results'])
+
+    # Ensure the cursor is properly depleted after one FT.CURSOR READ
+    iter = 0
+    while (cursor):
+        if (iter > 0):
+            env.debugPrint(f"iter:{iter}, n_received:{n_received}", force=True)
+        iter += 1
+        res, cursor = env.cmd('FT.CURSOR', 'READ', 'idx', cursor)
+        n_received += len(res['results'])
+
+        # Cursor should be depleted after the first read
+        env.assertEqual(cursor, 0, message=f"expected cursor to be depleted after one FT.CURSOR READ.")
+        env.assertEqual(len(res['results']), 0, message=f"expected to receive 0 results after one FT.CURSOR READ. iter:{iter}, n_received:{n_received}")
+
+    env.assertEqual(getCursorStats(env, 'idx')['index_total'], starting_cursor_count)
+
 def testCursorDepletionNonStrictTimeoutPolicy(env):
     """Tests that the cursor id is returned in case the timeout policy is
     non-strict (i.e., the `RETURN` timeout policy), even when a timeout is experienced"""
-
+    env = Env(protocol=3, moduleArgs='ON_TIMEOUT RETURN')
     conn = getConnectionByEnv(env)
-    run_command_on_all_shards(env, config_cmd(), 'SET', 'ON_TIMEOUT', 'RETURN')
 
     # Create the index
-    env.expect('FT.CREATE idx SCHEMA t text').ok()
+    env.expect('FT.CREATE idx SCHEMA t text sortable').ok()
```

Comment:
why sortable? missing `UNF`?

Reply:
just left overs of local tests i did.
Not related to UNF(?)